ui/pages/confirmations/hooks/alerts/useDomainMismatchAlert.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SignatureRequestType } from '../../types/confirm';
+import { isSIWESignatureRequest } from '../../utils';
+import useCurrentConfirmation from '../useCurrentConfirmation';
+
+const useDomainMismatchAlerts = (): Alert[] => {
+  const { currentConfirmation } = useCurrentConfirmation();
+  const t = useI18nContext();
+
+  const alerts = useMemo<Alert[]>(() => {
+    const isSIWE = isSIWESignatureRequest(
+      currentConfirmation as SignatureRequestType
+    );
+    if (!isSIWE || !currentConfirmation?.msgParams) {
+      return [];
+    }
+
+    const { msgParams } = currentConfirmation;
+    const isSIWEDomainValid = isValidSIWEOrigin(msgParams as WrappedSIWERequest);
+    if (isSIWEDomainValid) {
+      return [];
+    }
+
+    const alert = {
+      field: 'requestFrom',
+      key: 'requestFrom',
+      message: t('confirmAlertModalMessageDomainMismatch'),
+      reason: t('confirmAlertModalTitleSignIn'),
+      severity: Severity.Danger,
+    } as Alert;
+
+    return [alert];
```

Comment:
Minor, but could we return the alert directly rather than assigning to a variable first?

Reply:
can do [78363ec](https://github.com/MetaMask/metamask-extension/pull/25281/commits/78363ec074bf2ec07d9aa2045ebd6df9a0a76a52)